kube/src/service/url.rs

```diff
@@ -0,0 +1,56 @@
+use http::Request;
+use hyper::Body;
+
+/// Set cluster URL.
+pub fn set_cluster_url(req: Request<Body>, url: &url::Url) -> Request<Body> {
```

Comment:
This can be used for a (hacky) solution for #209 by adding a placeholder in `Api` then replacing it with `default_ns` before making a request.

```rust
pub fn default_namespaced(client: Client) -> Self {
    let resource = Resource::namespaced::<K>("{{default_ns}}");
    Self {
        resource,
        client,
        phantom: iter::empty(),
    }
}
```

```rust
let inner = ServiceBuilder::new()
    .map_request(move |r| finalize_url(r, &url, &default_ns))
    .service(client);
```

Reply:
Hah, that is actually a pretty good solution. The braces would mean we cannot clash with an actual namespace.